File: src/main/java/io/kuzzle/sdk/util/QueryObject.java
package io.kuzzle.sdk.util;

import org.json.JSONObject;

import io.kuzzle.sdk.listeners.ResponseListener;

/**
 * The type Query object.
 */
public class QueryObject {

  private JSONObject object;
  private String controller;
  private ResponseListener cb;

  /**
   * Gets object.
   *
   * @return the object
   */
  public JSONObject getObject() {
    return object;
  }

  /**
   * Sets object.
   *
   * @param object the object
   */
  public void setObject(JSONObject object) {
    this.object = object;
  }

  /**
   * Gets controller.
   *
   * @return the controller
   */
  public String getController() {
    return controller;
  }

  /**
   * Sets controller.
   *
   * @param controller the controller
   */
  public void setController(String controller) {
    this.controller = controller;
  }

  /**
   * Gets cb.
   *
   * @return the cb
   */
  public ResponseListener getCb() {
    return cb;
  }

  /**
   * Sets cb.
   *
   * @param cb the cb
   */
  public void setCb(ResponseListener cb) {
    this.cb = cb;
  }
}

File: src/main/java/io/kuzzle/sdk/util/QueryObject.kt
package io.kuzzle.sdk.util

import org.json.JSONObject

import java.util.Date

import io.kuzzle.sdk.core.Options
import io.kuzzle.sdk.listeners.OnQueryDoneListener

class QueryObject {
    /**
     * @return Query content
     */
    /**
     * @param query Query content
     */
    var query: JSONObject? = null
    /**
     * @return Controller action name to execute
     */
    /**
     * @param action Controller action name to execute
     */
    var action: String? = null
    /**
     * @return Query options
     */
    /**
     * @param options Query options
     */
    var options: Options? = null
    /**
     * @return Callback to invoke with the query result
     */
    /**
     * @param cb Callback to invoke with the query result
     */
    var cb: OnQueryDoneListener? = null
    /**
     * @return Query timestamp (Epoch time)
     */
    /**
     * @param timestamp Query timestamp (Epoch time)
     */
    var timestamp: Date? = null
}
